attachment_synchronize/models/attachment_synchronize_task.py

```diff
@@ -0,0 +1,241 @@
+
+    mako_template_env = SandboxedEnvironment(
+        variable_start_string="${",
+        variable_end_string="}",
+        line_statement_prefix="%",
+        trim_blocks=True,  # do not output newline after blocks
+    )
+    mako_template_env.globals.update(
+        {
+            "str": str,
+            "datetime": datetime,
+            "len": len,
+            "abs": abs,
+            "min": min,
+            "max": max,
+            "sum": sum,
+            "filter": filter,
+            "map": map,
+            "round": round,
+        }
+    )
+except ImportError:
+    _logger.warning("jinja2 not available, templating features will not work!")
+
+
+class AttachmentSynchronizeTask(models.Model):
+    _name = "attachment.synchronize.task"
+    _description = "Attachment synchronize task"
+
+    name = fields.Char(required=True)
+    method_type = fields.Selection(
+        [("import", "Import Task"), ("export", "Export Task")], required=True
+    )
+    pattern = fields.Char(
+        string="Selection Pattern",
+        help="Pattern used to select the files to be imported following the 'fnmatch' "
+        "special characters (e.g. '*.txt' to catch all the text files).\n"
+        "If empty, import all the files found in 'File Path'.",
+    )
+    filepath = fields.Char(
+        string="File Path", help="Path to imported/exported files in the Backend"
+    )
+    backend_id = fields.Many2one("fs.storage", string="Backend")
+    attachment_ids = fields.One2many("attachment.queue", "task_id", string="Attachment")
+    move_path = fields.Char(help="Imported File will be moved to this path")
+    new_name = fields.Char(
+        help="Imported File will be renamed to this name.\n"
+        "New Name can use 'mako' template where 'obj' is the original file's name.\n"
+        "For instance : ${obj.name}-${obj.create_date}.csv",
+    )
+    after_import = fields.Selection(
+        selection=[
+            ("rename", "Rename"),
+            ("move", "Move"),
+            ("move_rename", "Move & Rename"),
+            ("delete", "Delete"),
+        ],
+        help="Action after import a file",
+    )
+    file_type = fields.Selection(
+        selection=[],
+        help="Used to fill the 'File Type' field in the imported 'Attachments Queues'."
+        "\nFurther operations will be realized on these Attachments Queues depending "
+        "on their 'File Type' value.",
+    )
+    active = fields.Boolean("Enabled", default=True)
+    avoid_duplicated_files = fields.Boolean(
+        string="Avoid importing duplicated files",
+        help="If checked, a file will not be imported if an Attachment Queue with the "
+        "same name already exists.",
+    )
+    failure_emails = fields.Char(
+        help="Used to fill the 'Failure Emails' field in the 'Attachments Queues' "
+        "related to this task.\nAn alert will be sent to these emails if any operation "
+        "on these Attachment Queue's file type fails.",
+    )
```

Comment:
Hi @florian-dacosta Odoo introduced activities in 11.0 as internal communication system.
Clearly, users in odoo do not use it as much as they could do.
Flows in this module are fully managed by Odoo objects but mail is a communication mode to exchange informations between two  systems. Here we have a unique system, `Odoo`
Currently our odoo users in our projects complains they receive too many mails in a day from misc sources for so many usage.
I don't talk about environmental considerations, but we could have this argument too
I think we should promote definitively this kind of communication in a such version.
Replace here `failure_emails` here by `failure_activity_user_ids` could definitively imply users to use activities instead of emails.
It's an opportunity to improve communication, make it smarter, interactive, only few bytes consumer

#maketheplanetgreenagain

My 2 cts

Reply:
Hi @bealdav 
Thanks for the remark. It is true we may want to implement activities.
This module is not responsible for this part though, as it just extends it, but the base implementation of these failure email is in the base module (`attachment_queue`) which is merged.

I think it would be nice to have an implementation with the activities indeed, and we could have another one for those still prefering email.
We'd just have to decide if we'd implement both in base module, or only activities and email as a separated module, or both in separated modules...

Overall I agree,  but I think it should be put on the roadmap. (I may work on it but I don't want it to block the PR)